tests/test_create.py

```diff
@@ -2177,104 +2177,117 @@ def side_effect(self, url, **kwargs):
 
-            assert any(isinstance(e, RemoveError) for e in exc.value.errors)
-            assert package_is_installed(prefix, "conda")
-            assert package_is_installed(prefix, "pycosat")
+        with pytest.raises(CondaMultiError) as exc:
+            conda_cli("remove", f"--prefix={prefix}", "pycosat", "--yes")
 
-            with pytest.raises(CondaMultiError) as exc:
-                run_command(Commands.REMOVE, prefix, "conda")
+        assert any(isinstance(e, RemoveError) for e in exc.value.errors)
+        assert package_is_installed(prefix, "conda")
+        assert package_is_installed(prefix, "pycosat")
 
-            assert any(isinstance(e, RemoveError) for e in exc.value.errors)
-            assert package_is_installed(prefix, "conda")
-            assert package_is_installed(prefix, "pycosat")
+        with pytest.raises(CondaMultiError) as exc:
+            conda_cli("remove", f"--prefix={prefix}", "conda", "--yes")
 
+        assert any(isinstance(e, RemoveError) for e in exc.value.errors)
+        assert package_is_installed(prefix, "conda")
+        assert package_is_installed(prefix, "pycosat")
 
-def test_force_remove():
-    with make_temp_env() as prefix:
-        stdout, stderr, _ = run_command(Commands.INSTALL, prefix, "libarchive")
+
+def test_force_remove(
+    tmp_env: TmpEnvFixture, conda_cli: CondaCLIFixture, monkeypatch: MonkeyPatch
+):
+    with tmp_env("libarchive") as prefix:
         assert package_is_installed(prefix, "libarchive")
         assert package_is_installed(prefix, "xz")
 
-        stdout, stderr, _ = run_command(Commands.REMOVE, prefix, "xz", "--force")
+        conda_cli("remove", f"--prefix={prefix}", "xz", "--force", "--yes")
         assert not package_is_installed(prefix, "xz")
         assert package_is_installed(prefix, "libarchive")
 
-        stdout, stderr, _ = run_command(Commands.REMOVE, prefix, "libarchive")
+        conda_cli("remove", f"--prefix={prefix}", "libarchive", "--yes")
         assert not package_is_installed(prefix, "libarchive")
 
     # regression test for #3489
     # don't raise for remove --all if environment doesn't exist
+    # split this into a new test
     rm_rf(prefix, clean_empty_parents=True)
-    run_command(Commands.REMOVE, prefix, "--all")
+    conda_cli("remove", f"--prefix={prefix}", "--all")
 
 
-def test_download_only_flag():
+def test_download_only_flag(
+    tmp_env: TmpEnvFixture, mocker: MockerFixture, conda_cli: CondaCLIFixture
+):
     from conda.core.link import UnlinkLinkTransaction
 
-    with patch.object(UnlinkLinkTransaction, "execute") as mock_method:
-        with make_temp_env("openssl", "--download-only", use_exception_handler=True):
-            assert mock_method.call_count == 0
-        with make_temp_env("openssl", use_exception_handler=True):
-            assert mock_method.call_count == 1
+    with tmp_env() as prefix:
+        spy = mocker.spy(UnlinkLinkTransaction, "execute")
```

Comment:
Neat! (`mocker.spy` 👀 )

Reply:
yup! the problem with the standard mock methods is they by default will modify the mocked thing's behavior, `spy` is great if all we need are stats